### PR TITLE
Documentation: document map parsing design

### PR DESCRIPTION
**Pull request by DanVanAtta**
Documents the discussion started in: https://github.com/triplea-game/triplea/issues/7498,
covers the terms parsing and game data assembly and describes the design
philosophy that map-reading should just read the contents of XML and assembly
converts the XML model objects into actual java game objects.

### CODE REVIEW

**Comment by Cernelius** [comment]
Since even a developer misunderstood what I wrote down, chances are that I'm not being very clear. Unfortunately, I don't know how to present it better, but I'm open to suggestions (just someone tell me how it would reword it and I can at least check the substance is not adulterated under my understanding).

If it creates more confusion than help, the "Note" section can be completely removed (it just reproposes items, thus it is unnecessary). Just I made it because I assume otherwise people will assume that if their maps work after updating that means they updated it correctly (which, sadly, it's not the case, as the program doesn't fully validate everything).

**Comment by DanVanAtta** [comment]
@Cernelius wrong PR on your last comment : )

See my last few comments in: https://github.com/triplea-game/triplea/pull/7630, I think the trick is to convert the document into a legend of deprecated/old items and what their latest should be

**Review by RoiEXLab — APPROVED** [review]
